### PR TITLE
Added TS, SCSS, Unit Testing, and a sample component

### PR DESCRIPTION
**Pull request by JacobN256**
- Generated project using VueCLI
- Added .vscode extensions recommendation
- Created `Counter` component using TS
- Added `counter.spec.ts` with a few tests

**Notes:**
- Pending e2e testing verification

### CODE REVIEW

File: src/components/counter/counter.vue [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<style lang="scss">
+@import './counter.scss';
```
Comment (prowe214) [inline src/components/counter/counter.vue:2]: 👍I like this better than the other two patterns of styles I was playing with

File: src/router.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      path: '/',
+      name: 'home',
+      component: Home,
+    },
+    {
+      path: '/about',
+      name: 'about',
+      // route level code-splitting
+      // this generates a separate chunk (about.[hash].js) for this route
+      // which is lazy-loaded when the route is visited.
```
Comment (prowe214) [inline src/router.ts:19]: Nice, didn't know about this!

**Review by chrisman — COMMENTED** [review]
This appears to have removed the webpack config. How is the build process being managed now? Maybe in tsconfig.json?

**Review by prowe214 — APPROVED** [review]
Looking good to me!  

For knowledge sharing -- since this is a big PR -- pick out a couple of core pieces to explain, so we can review the work as a team on our next meeting (maybe 10min) and teach each other.  Try to also dive into what the CLI did, so that we can learn how to make the decision of when to use the CLI vs when we can wire it up ourselves.  

Nice job! 😄

File: src/router.ts (hunk 2 above)
Comment (JacobN256) [reply src/router.ts:19]: I didn't either tbh, that was added when I selected the `router` option in the VueCLI

**Comment by JacobN256** [comment]
> This appears to have removed the webpack config. How is the build process being managed now? Maybe in tsconfig.json?

From my understanding, when I configure the VueCLI it gives me 2 options on project configuration.  Seperate configuration files for each sub-system, or it throws all the config info in the `package.json` for Jest, tslint, Css post processor. (I felt like all in one, is cleaner.)

For webpack, the Vue-Cli-Service is standing up a development server and configuring webpack for us.  We can modify the webpack settings two different ways, we can add a `vue.config.js` file or we can add a `vue` tag in the `package.json` and setup configuration info there.  (pushing an update with an outputDir set) . Check out this link for more info on config info.

https://cli.vuejs.org/config/#vue-config-js

**Pull request merged**